packages/core/src/components/Search/Search.tsx

```diff
@@ -1,124 +1,127 @@
-import { NOOP } from "../../utils/function-utils";
-import { SubIcon, VibeComponentProps, VibeComponent, withStaticProps } from "../../types";
-import { TextFieldTextType } from "../TextField/TextFieldConstants";
-import { BASE_SIZES } from "../../constants";
 import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
 import styles from "./Search.module.scss";
+import BaseInput from "../BaseInput/BaseInput";
+import useDebounceEvent from "../../hooks/useDebounceEvent";
+import IconButton from "../IconButton/IconButton";
+import Icon from "../Icon/Icon";
+import { SearchComponent } from "./Search.types";
+import Loader from "../Loader/Loader";
 
-export interface SearchProps extends VibeComponentProps {
-  secondaryIconName?: SubIcon;
-  iconName?: SubIcon;
-  onChange?: (value: string) => void;
-  autoFocus?: boolean;
-  value?: string;
-  placeholder?: string;
-  disabled?: boolean;
-  debounceRate?: number;
-  onBlur?: (event: React.FocusEvent) => void;
-  onFocus?: (event: React.FocusEvent) => void;
-  wrapperClassName?: string;
-  setRef?: () => void;
-  autoComplete?: string;
-  /* BASE_SIZES is exposed on the component itself */
-  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
-  /* TYPES is exposed on the component itself */
-  type?: SearchType;
-  validation?:
-    | {
-        status: "error" | "success";
-        text: string;
-      }
-    | { text: string };
-  inputAriaLabel?: string;
-  searchResultsContainerId?: string;
-  activeDescendant?: string;
-  /*  Icon names labels for a11y */
-  iconNames?: {
-    layout: string;
-    primary: string;
-    secondary: string;
-  };
-  /** shows loading animation */
-  loading?: boolean;
-}
-
-const Search: VibeComponent<SearchProps, unknown> & {
-  sizes?: typeof BASE_SIZES;
-  types?: typeof SearchType;
-} = forwardRef<unknown, SearchProps>(
+const Search: SearchComponent = forwardRef(
   (
     {
-      secondaryIconName = CloseIcon,
-      iconName = SearchIcon,
-      onChange = NOOP,
-      autoFocus = false,
-      value = "",
-      placeholder = "",
-      disabled = false,
-      debounceRate = 200,
-      onBlur = NOOP,
-      onFocus = NOOP,
-      wrapperClassName = "",
-      setRef = NOOP,
+      searchIconName = SearchIcon,
+      clearIconName = CloseSmallIcon,
+      clearIconLabel = "Clear",
+      additionalActionRender: AdditionalAction,
+      value,
+      placeholder,
+      size = "medium",
+      disabled,
+      loading,
+      autoFocus,
       autoComplete = "off",
-      size = BASE_SIZES.MEDIUM,
-      type = SearchType.SQUARE,
-      className,
-      id = "search",
-      validation = null,
       inputAriaLabel,
-      searchResultsContainerId = "",
-      activeDescendant = "",
-      iconNames = SearchDefaultIconNames,
-      loading = false,
+      debounceRate = 200,
+      searchResultsContainerId,
+      activeDescendant,
+      onChange,
+      onFocus,
+      onBlur,
+      wrapperClassName,
+      className,
```

Comment:
it would prevent people from editing the input
do we really want it?
should we remove it and it only if requested?

should we just call it className or wrapperClassName?

Reply:
I think that the className we provide should be **always** on the outer element, because otherwise if users give it spacing values or something similar they will get unexpected behaviour. I think it should be the other way around - `className` on the outer element, and `inputClassName` on the inner element, and like you said, only if someone asks for it because I don't see any good reason to change any css properties for it